src/dawn/Optimizer/OptimizerContext.cpp

```diff
@@ -219,13 +235,14 @@ class StencilDescStatementMapper : public iir::ASTVisitor {
     // Remove the nested VerticalRegionDeclStmts and StencilCallDeclStmts
     RemoveStencilDescNodes remover;
     for(auto& statement : scope_.top()->controlFlowDescriptor_.getStatements())
-      statement->ASTStmt->accept(remover);
+      statement->accept(remover);
   }
 
   /// @brief Push back a new statement to the end of the current statement list
   void pushBackStatement(const std::shared_ptr<iir::Stmt>& stmt) {
-    scope_.top()->controlFlowDescriptor_.insertStmt(
-        std::make_shared<Statement>(stmt, scope_.top()->StackTrace));
+    stmt->getData<iir::IIRStmtData>().StackTrace =
+        boost::optional<std::vector<ast::StencilCall*>>(scope_.top()->StackTrace);
```

Comment:
There should be a `make_optional` which probably fits better?

Reply:
I see uses of both `make_optional` and `boost::optional`'s constructor in other parts of the code ... what's the difference?